File: packages/ios/Sources/Components/NumberInput/NumberInput.figma.swift
import SwiftUI
import Figma

struct KozmosNumberInputConnect: FigmaConnect {
    let component = KozmosNumberInput.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=338-1796"

    @FigmaString("Label Text")
    var label: String = "Number"

    @FigmaString("Value Text")
    var valueText: String = "12"

    @FigmaString("Placeholder Text")
    var placeholder: String = "0"

    @FigmaString("Helper Text")
    var helperText: String = "Use a whole number."

    @FigmaBoolean("Show Helper Text")
    var showHelperText: Bool = true

    @FigmaEnum(
        "State",
        mapping: [
            "Default": "default",
            "Focus": "focus",
            "Disabled": "disabled",
            "Readonly": "readonly"
        ]
    )
    var state: String = "default"

    @FigmaEnum(
        "Status",
        mapping: [
            "Default": KozmosInputStatus.`default`,
            "Error": KozmosInputStatus.error,
            "Warning": KozmosInputStatus.warning,
            "Success": KozmosInputStatus.success
        ]
    )
    var status: KozmosInputStatus = .default

    @FigmaEnum(
        "Steppers",
        mapping: [
            "True": true,
            "False": false
        ]
    )
    var showSteppers: Bool = true

    var body: some View {
        KozmosNumberInput(
            value: .constant(Double(self.valueText)),
            label: self.label,
            placeholder: self.placeholder,
            disabled: self.state == "disabled",
            readOnly: self.state == "readonly",
            status: self.status,
            helperText: self.showHelperText ? self.helperText : nil,
            showSteppers: self.showSteppers
        )
    }
}

File: packages/android/src/main/java/com/kozmos/components/NumberInput/NumberInput.figma.kt
package com.kozmos.components.numberinput

import androidx.compose.runtime.Composable
import com.figma.code.connect.Figma
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType
import com.kozmos.components.input.KozmosInputStatus

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=338-1796")
class KozmosNumberInputConnect {
    @FigmaProperty(FigmaType.Text, "Label Text")
    val label: String = "Number"

    @FigmaProperty(FigmaType.Text, "Value Text")
    val valueText: String = "12"

    @FigmaProperty(FigmaType.Text, "Placeholder Text")
    val placeholder: String = "0"

    @FigmaProperty(FigmaType.Text, "Helper Text")
    val helperText: String = "Use a whole number."

    @FigmaProperty(FigmaType.Boolean, "Show Helper Text")
    val showHelperText: Boolean = true

    @FigmaProperty(FigmaType.Enum, "State")
    val state: String = Figma.mapping(
        "Default" to "default",
        "Focus" to "focus",
        "Disabled" to "disabled",
        "Readonly" to "readonly"
    )

    @FigmaProperty(FigmaType.Enum, "Status")
    val status: KozmosInputStatus = Figma.mapping(
        "Default" to KozmosInputStatus.Default,
        "Error" to KozmosInputStatus.Error,
        "Warning" to KozmosInputStatus.Warning,
        "Success" to KozmosInputStatus.Success
    )

    @FigmaProperty(FigmaType.Enum, "Steppers")
    val showSteppers: Boolean = Figma.mapping(
        "True" to true,
        "False" to false
    )

    @Composable
    fun ComponentExample() {
        KozmosNumberInput(
            value = valueText.toDoubleOrNull(),
            onValueChange = {},
            label = label,
            placeholder = placeholder,
            enabled = state != "disabled",
            readOnly = state == "readonly",
            status = status,
            helperText = if (showHelperText) helperText else null,
            showSteppers = showSteppers
        )
    }
}
